allennlp/modules/vision/region_detector.py

```diff
@@ -37,194 +52,238 @@ class RandomRegionDetector(RegionDetector):
     """
 
     def forward(
-        self, raw_images: FloatTensor, image_sizes: IntTensor, featurized_images: FloatTensor
-    ) -> Dict[str, torch.Tensor]:
-        batch_size, num_features, height, width = raw_images.size()
-        features = torch.rand(batch_size, 2, 10, dtype=featurized_images.dtype).to(
-            raw_images.device
-        )
-        coordinates = torch.zeros(batch_size, 2, 4, dtype=torch.float32).to(raw_images.device)
+        self,
+        images: FloatTensor,
+        sizes: IntTensor,
+        image_features: "OrderedDict[str, FloatTensor]",
+    ) -> RegionDetectorOutput:
+        batch_size, num_features, height, width = images.size()
+        features = [
+            torch.rand(2, 10, dtype=torch.float32, device=images.device) for _ in range(batch_size)
+        ]
+        boxes = [
+            torch.zeros(2, 4, dtype=torch.float32, device=images.device) for _ in range(batch_size)
+        ]
         for image_num in range(batch_size):
-            coordinates[image_num, 0, 2] = image_sizes[image_num, 0]
-            coordinates[image_num, 0, 3] = image_sizes[image_num, 1]
-            coordinates[image_num, 1, 2] = image_sizes[image_num, 0]
-            coordinates[image_num, 1, 3] = image_sizes[image_num, 1]
-        return {"features": features, "coordinates": coordinates}
+            boxes[image_num][0, 2] = sizes[image_num, 0]
+            boxes[image_num][0, 3] = sizes[image_num, 1]
+            boxes[image_num][1, 2] = sizes[image_num, 0]
+            boxes[image_num][1, 3] = sizes[image_num, 1]
+        return RegionDetectorOutput(features, boxes)
 
 
 @RegionDetector.register("faster_rcnn")
 class FasterRcnnRegionDetector(RegionDetector):
```

Comment:
Not in this PR, but it would be good if we cached these somewhere/somehow. In a multi-task setting, multiple readers will be featurizing images in the same way, and they don't all need to instantiate their own frozen models to do so.